uavcan: publish CAN interface status as uorb topic

oystub opened the pull request:
### Solved Problem
While working on https://github.com/PX4/PX4-Autopilot/issues/22871 we discovered that there is no existing method for logging can interface errors and tx/rx statistics.

As UAVCAN peripherals become more widely used, it is important to be able to properly evaluate CAN bus performance. For us this is useful both:
- Preventatively, in order to catch faulty peripherals/wiring before a serious problem happens
- As an important aid when diagnosing flight logs from errors/accidents. For example after GPS loss incidents or motor/servo control issues

### Solution
Created a new uorb topic,  `can_interface_status`, which is published at 10 Hz, and included in the standard logged topics.

### Changelog Entry
For release notes:
```
Feature: CAN interface status is published as uorb topic
```

### Alternatives
It could also be possible to add additional performance counters to the uavcan module, and compare values at start of flight and end. However, this only tells IF there are errors and not when they occured. This is highly relevant when using the log as a diagnostic tool, in order to to see if CAN errors are related to a problem. For example, it is usual to have some CAN bus errors during boot/initialization. In the event of crashes/log streaming, it is also very useful to get the data streamed, instead of relying on saving the perf counters at end of log.

### Test coverage
Tested on a CubeOrange with two Here4 GPSes connected, one to each CAN port. Using a perf counter, the time used to check status and publish was measured.
```
if (hrt_absolute_time() - _last_can_status_pub >= interval) {
	perf_begin(_status_publish);
	...
        ...
	perf_end(_status_publish);
}
```
which gave:
`status publish time: 5571 events, 37134us elapsed, 6.67us avg, min 4us max 575us 21.531us rms`

Additional time used for publishing this info is reasonable to default to publishing at 10 Hz. (I suspect the perf counter itself is a significant contributor to time used here)

dagar commented on msg/CanInterfaceStatus.msg:
```diff
@@ -0,0 +1,5 @@
+uint64 timestamp		# time since system start (microseconds)
+
```
Should this be per bus?

oystub replied:
My thought was that is enough to set it up as multi-instance topic, with the instance number mirroring the bus index. But we could probably add a `bus_index` or `bus_id`, like with `device_id` for a lot of other sensors.

oystub commented:
Removed the configurable rate parameter, I don't think it is really needed. Given the performance, it is no problem to log at 10 Hz, and i don't see any advantage of logging at a higher rate. I think it is better to just keep it simple.

Fixed a bug where the interface index from orb_advertise_multi would overwrite the iteration variable `i`.

Rebased onto main.

oystub commented:
I also looked at making this status multi-topic instead of multi-instance, as discussed in the Community Q&A, but this would involve hard-coding the number of CAN interfaces in the message definition, like for `ActuatorControlsStatus`. I think a multi-instance topic which supports arbitrary number of interfaces is best. 
Do anyone have different opinions on this?

MaEtUgR commented on src/modules/logger/logged_topics.cpp:
```diff
@@ -53,6 +53,7 @@ void LoggedTopics::add_default_topics()
 	add_optional_topic("autotune_attitude_control_status", 100);
 	add_optional_topic("camera_capture");
 	add_optional_topic("camera_trigger");
+	add_optional_topic("can_interface_status", 10);
```
```suggestion
	add_optional_topic("can_interface_status", 100);
```
Probably should be 10Hz -> 100ms intervals which is the definition of this parameter:
https://github.com/PX4/PX4-Autopilot/blob/710286da722beae72a488af52b6c1e603f68eea7/src/modules/logger/logged_topics.h#L116

oystub replied:
Sorry, just saw this now. 

This should actually be `add_optional_topic_multi("can_interface_status", 100);`, to get the multiple instances right. My bad.

Can you commit it directly, @dagar, or do you want me to open a new pr?

oystub commented:
There are still some improvements to make here, so I will post a new pr, addressing:
- logging rate
- advertising the can status topics before the logger is started, so they are properly logged also when not subscribed to elsewhere.
- use cpp style publishers properly.

However, while testing this, randomly I noticed that running
```
uavcan stop
uavcan start
```
on `release/1.14` or later, crashes the CubeOrange (not on `release/1.13`). So I will try to figure that out at the same time.
